src/main/java/com/users/service/UserService.java

```diff
@@ -0,0 +1,75 @@
+package com.users.service;
+
+import com.users.dto.UserDto;
+import com.users.entities.User;
+import com.users.exceptions.EntityAlreadyExistsException;
+import com.users.exceptions.EntityNotFoundException;
+import com.users.mappers.UserMapper;
+import com.users.repository.UserRepository;
+import com.users.util.AdultValidator;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Transactional
+@AllArgsConstructor
+@Service
+@Slf4j
+public class UserService {
+
+    private UserRepository userRepository;
+    private UserMapper userMapper;
+
+    public UserDto create(UserDto user) {
+        log.debug("User to be created {}", user);
+        if (!AdultValidator.validateDateOfBirth(user.getDateOfBirth())) {
+            throw new IllegalArgumentException("User must be 18 years old");
+        }
+        if (userRepository.existsByEmail(user.getEmail())) {
+            throw new EntityAlreadyExistsException(String.format("User with email %s already exists", user.getEmail()));
+        }
+        User userEntity = userMapper.toEntity(user);
+        User savedUser = userRepository.save(userEntity);
+        return userMapper.toDto(savedUser);
+    }
+
+    public UserDto updateUser(Long id, UserDto userDto) {
+        log.debug("Update user with id {}", id);
+        if (!AdultValidator.validateDateOfBirth(userDto.getDateOfBirth())) {
+            throw new IllegalArgumentException("User must be 18 years old");
+        }
+        User user = findById(id);
+        userMapper.updateEntity(user, userDto);
+        User savedUser = userRepository.save(user);
+        return userMapper.toDto(savedUser);
+    }
+
+    public List<UserDto> findAllUsers() {
+        log.debug("Request to get all users");
+        List<User> allUsers = userRepository.findAll();
+        return allUsers.stream().map(userMapper::toDto).collect(Collectors.toList());
```

Comment:
poti sa creezi in mapper o metoda care sa primeasca o lista de User si sa returneze o lista de DTO uri

Reply:
banuiesc ca voi face asa in mapper:
ist<UserDto> toDto(List<User> users);
iar in findAll asa:
 public List<UserDto> findAllUsers() {
        log.debug("Request to get all users");
        List<User> allUsers = userRepository.findAll();
        return userMapper.toDto(allUsers);